data/common.yaml

```diff
@@ -20,6 +23,13 @@ mcollective::server_config:
   plugin.yaml: "/etc/puppetlabs/mcollective/generated-facts.yaml"
   registerinterval: 10
   registration: "Choria"
+  logfile: "/var/log/puppetlabs/mcollective.log"
```

Comment:
the pattern of properties and hashes is widely used for good reason - for example in the official mysql module see `override_options`.

With your proposal, please answer:

  * How do we validate the key properties are the right type and form?
  * How do we communicate to users which are available? Is there a facility for standard introspection to learn about these?
  * How do we share key items like where the libdir is across classes and modules?